samples/bluetooth/encrypted_advertising/central/src/central_ead.c

```diff
@@ -0,0 +1,526 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdint.h>
+#include <string.h>
+
+#include <zephyr/kernel.h>
+
+#include <zephyr/net/buf.h>
+#include <zephyr/sys/printk.h>
+#include <zephyr/logging/log.h>
+
+#include <zephyr/bluetooth/ead.h>
+#include <zephyr/bluetooth/att.h>
+#include <zephyr/bluetooth/gatt.h>
+#include <zephyr/bluetooth/addr.h>
+#include <zephyr/bluetooth/conn.h>
+#include <zephyr/bluetooth/uuid.h>
+#include <zephyr/bluetooth/bluetooth.h>
+
+#include "common/bt_str.h"
+
+#include "common.h"
+
+LOG_MODULE_REGISTER(ead_central_sample, CONFIG_BT_EAD_LOG_LEVEL);
+
+static uint8_t *received_data;
```

Comment:
I suggest using a `net_buf_simple` to manage the offset. We should also properly check the capacity before adding data to it.

```suggestion
static net_buf_simple *received_data;
```